### PR TITLE
Feature/api calls

### LINKED ISSUE #9: Add fetch request to use api data instead of mock

### LINKED ISSUE #13: add error handling for 500 error

### PR DESCRIPTION
**Pull request by mcraig2342**
# Pull Request Template
## Description
closes #9 closes #13 
Please include a summary of the change and which issue is fixed.

Movie data is now being fetched from the api instead of using mock data. Added error handling for 500 errors so that a message is displayed on the dom if the movies cannot be found. When a user clicks on a movie, the corresponding video data is fetched and stored in state.

Please include which files need to be reviewed.

app.js and api-calls.js

Please also include relevant screenshots of your code being deployed. 

Screen Shot 2021-05-15 at 12 46 22 PM

Fixes # (issue)
## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,35 @@
...
+    .catch(error => this.setState({ error: 'Something went wrong!'}))
   }
 
   showMovieDetails = event => {
     this.setState({currentMovie: event.target.id})
+    getVideo(event.target.id)
+    .then(video => {
+      this.setState({currentVideo: video})
+    })
+    .catch(error => this.setState({ error: 'Video is unavailable'}))
```
Comment (AndieDrew) [inline src/App.js:32]: Now that we have this waiting for further functionality we can plan how we'd like to display the videos. Perhaps we could display the first video of the videos array, and have buttons on left and right to iterate through the videos array by one index so user can click through all videos associated with a movie.

File: src/App.js [hunk 2]
```diff
@@ -29,10 +40,12 @@ class App extends Component {
     return (
       <main className = 'App'>
         <Header currentState={this.state.currentMovie}/>
-        {!this.state.currentMovie &&
+        {this.state.error &&
+         <h1>Sorry no movies found 🤷‍♂️</h1>}
+        {!this.state.currentMovie && !this.state.error &&
```
Comment (AndieDrew) [inline src/App.js:45]: Now that we have the functionality of error feedback being displayed to user we can spend the rest of our weekend refactoring.

File: src/api-calls.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export const getMovie = () => {
+  return fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
+    .then(response => response.json())
+}
+
+export const getVideo = ( id ) => {
+  return fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${id}/videos`)
+    .then(response => response.json())
+}
```
Comment (AndieDrew) [inline src/api-calls.js:9]: Simple and clean, I don't think we'd need to write this any other way.

File: src/App.js [hunk 4]
```diff
@@ -29,10 +40,12 @@ class App extends Component {
...
           <Area className='area' movies={this.state.movies} details={this.showMovieDetails}/>
         }
-        {this.state.currentMovie &&
+        {this.state.currentMovie && !this.state.error &&
```
Comment (AndieDrew) [inline src/App.js:48]: Since this will prevent the Details background styling from displaying we should add some styling for our error container.

**Review by AndieDrew — APPROVED** [review]
Everything looks good. Excited to spend our extra time refactoring, adding some styling and a video display feature.

File: src/App.js [hunk 5]
```diff
@@ -1,24 +1,35 @@
...
       movies: [],
       currentMovie: null,
+      currentVideo: null,
+      error: '',
     };
   }
 
   componentDidMount() {
-    this.setState({movies: movieData.movies})
+    getMovie()
```
Comment (sertmer) [inline src/App.js:19]: The function is called getMovie (singular) but appears to get all movies? I would consider changing that to match.

File: src/App.js (hunk 4 above)
Comment (sertmer) [reply src/App.js:48]: Yes! Or - consider building a reusable Error component? Could have a little error image, and maybe a button to go back home and try again? Up to you!

File: src/api-calls.js (hunk 3 above)
Comment (sertmer) [inline src/api-calls.js:9]: What about if the fetch fails? might be good to add some sad path operations here. Additionally - these fetches look nearly identical - can you think of any way to DRY them up into one, using args and params?

**Pull request merged**